### PR TITLE
lib/config: reasonable default config values

### PR DESCRIPTION
**Pull request by xhebox**
Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Issue Number: ref #161 

Problem Summary: Update default config values. This will enable to write less code in tidb-operator and dedicated tier.

What is changed:

1. default values for all config options.
2. tiproxy does not need a config file to start anymore.
3. Remove useless `rsa-key-size` in the security section.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
```
rm conf/proxy.toml
./bin/tiproxy
# this could start directly now
```
- [ ] No code

Notable changes

- [ ] Has configuration change
- [ ] Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- [ ] Has tiproxyctl change
- [ ] Other user behavior changes

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: lib/util/security/tls.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ import (
 	"go.uber.org/zap"
 )
 
-const DefaultCertExpiration = 10 * 365 * 24 * time.Hour
+const DefaultCertExpiration = 72 * time.Hour
```
Comment (djshow832) [inline lib/util/security/tls.go:37]: Why 72 hours?

Comment (xhebox) [reply lib/util/security/tls.go:37]: The original default value will be trouble some to represent in `hours`. And the maximum duration unit is  `hr`.

The expiration time will be extremely for 10 years, which is scary on the first sight of `proxy.toml`.

Considering, you can now specify the config by yourself and auto-certs should not be used in serious cases, it is OK to me.

Comment (djshow832) [reply lib/util/security/tls.go:37]: We need a reasonable value for default values, otherwise it's senseless. I think 3 months is a reasonable time.

Comment (xhebox) [reply lib/util/security/tls.go]: Makes sense to me,

**Review by djshow832 — APPROVED** [review]

**Pull request merged**